src/bin/inx-chronicle/api/stardust/core/routes.rs

```diff
@@ -308,14 +332,33 @@ async fn transaction_included_block(
         .await?
         .ok_or(ApiError::NoResults)?;
 
-    Ok(BlockResponse::Json(BlockDto::try_from(block)?))
+    let protocol_params = database
+        .collection::<ProtocolUpdateCollection>()
+        .get_protocol_parameters_for_version(block.protocol_version)
+        .await?
+        .ok_or(ApiError::NoResults)?
+        .parameters
+        .try_into()?;
+
+    Ok(BlockResponse::Json(BlockDto::try_from_with_context(
+        &protocol_params,
+        block,
+    )?))
 }
 
 async fn receipts(database: Extension<MongoDb>) -> ApiResult<ReceiptsResponse> {
     let mut receipts_at = database.collection::<MilestoneCollection>().get_all_receipts().await?;
     let mut receipts = Vec::new();
     while let Some((receipt, at)) = receipts_at.try_next().await? {
-        let receipt: &bee_block_stardust::payload::milestone::MilestoneOption = &receipt.try_into()?;
+        let protocol_params = database
```

Comment:
I wrote it this way, but I strongly dislike it because we fetch protocol params for every receipt. Can we find a better way or create an issue to fix it later at least?

Reply:
Yeah, please create an issue.